### PR TITLE
Tranche advanced settings

### LINKED ISSUE #210: app issue

### PR DESCRIPTION
**Pull request by BartDeCaluwe**
This PR will add:

- edit tranche drawer where you can enable/disable early withdrawals for interest and principal
- Advanced tranche settings on the review page
- updated default tranche values

## Checklist

- [x] I used the [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/) specification.
- [ ] Unit tests have been created if a new feature was added.
- [x] Translations have been added and extracted if a new feature was added.

### CODE REVIEW

**Comment by RuvenSalamon** [comment]
conflicts

**Comment by RuvendeGroote** [comment]
conflicts

File: src/components/DealForm.tsx [hunk 1]
```diff
@@ -5,11 +5,11 @@ import { DealDetailsStep } from "@components/DealDetailsStep";
 import { DealTranchesStep } from "@components/DealTranchesStep";
 import { ReviewDealStep } from "@components/ReviewDeal";
 import { defineMessages, useIntl } from "react-intl";
-import { threeTrancheStructure } from "@consts";
+import { defaultTranches } from "@consts";
 import { newDealDefaults } from "@consts";
 
 const dealFormDefaultValues = {
-	trancheStructure: threeTrancheStructure.value,
+	trancheStructure: defaultTranches[defaultTranches.length - 1].value,
```
Comment (RuvendeGroote) [inline src/components/DealForm.tsx:13]: the values of these defaultTranches are equal to the trancheSettingsFields
defaulttranches.value is used as the key of a form item at the same time we refer to those form items using the trancheSettingsFields const. I want to unify those.

**Review by RuvendeGroote — CHANGES_REQUESTED** [review]
wip review (needed to review something else) but you can already fix the comment

File: src/components/FormItem.tsx [hunk 2]
```diff
@@ -86,9 +86,10 @@ export const FormItem = ({
 	}
 
 	const className = classNames([
-		"font-bold text-base mb-7",
+		"font-bold text-base",
 		disabled && "text-neutral-60/40",
 		hasFeedback && validateStatus === "error" && "border-error",
+		labelClassName && !labelClassName.includes("mb-") && "mb-7",
```
Comment (RuvendeGroote) [inline src/components/FormItem.tsx]: doesn't this always either return mb-7 or falsy?
and why is it not allowed for the label to have a mb- css class?

File: src/components/ReviewDeal.tsx [hunk 3]
```diff
@@ -198,23 +207,41 @@ export const ReviewDealStep: FunctionComponent<ReviewDealStepProps> = ({
...
+					title={trancheTitleMap[trancheStructure]}
 					checked={false}
 					isInteractive={false}
 					showContent={true}
 					className="col-span-4"
 				/>
 			)}
+			{formTranchesWithAdvancedSettings.length > 0 && (
+				<div className="uppercase text-2xl font-bold">
+					{intl.formatMessage({
```
Comment (RuvendeGroote) [inline src/components/ReviewDeal.tsx]: put messages separate

File: src/components/TrancheLine.tsx [hunk 4]
```diff
@@ -1,22 +1,22 @@
...
+import Big from "big.js";
 import React, { FunctionComponent } from "react";
 
 interface TrancheLineProps {
 	name: string;
 	highlightedElement: string;
 	color: string;
-	apr: Fraction;
-	value: number;
+	structure: TrancheSettings;
```
Comment (RuvendeGroote) [inline src/components/TrancheLine.tsx]: rename. these are settings and we also have a tranchestructure type so this is confusing

File: src/components/TrancheLine.tsx [hunk 5]
```diff
@@ -37,13 +37,33 @@ export const TrancheLine: FunctionComponent<TrancheLineProps> = ({
...
+				{percentageOfInterest
+					? ratioFormatter.format(
+							Big(percentageOfInterest.toString().replace(",", "")).div(100).toNumber()
+					  )
+					: "/"}
+			</div>
+			<div className={getTextClassNames()}>
+				{" "}
+				{apr
+					? ratioFormatter.format(Big(apr.toString().replace(",", "")).div(100).toNumber())
```
Comment (RuvendeGroote) [inline src/components/TrancheLine.tsx]: not sure if this always works? isn't this dependent on locale? also, can we abstract this as we are doing the same thing multiple times.

File: src/components/TrancheOption.tsx [hunk 6]
```diff
@@ -20,44 +20,69 @@ export const TrancheOption: FunctionComponent<TrancheOptionProps> = ({ trancheDa
...
-	const donut = useMemo(
-		() => (
+	const donut = useMemo(() => {
+		if (!trancheStructure) {
+			return null;
+		}
+
+		const tranches = [
+			{
+				name: "Senior",
```
Comment (RuvendeGroote) [inline src/components/TrancheOption.tsx]: can we have a tranche name enum or something somewhere?

**Review by RuvendeGroote — CHANGES_REQUESTED** [review]

File: src/components/DealForm.tsx (hunk 1 above)
Comment (RuvendeGroote) [reply src/components/DealForm.tsx:13]: was this fixed? I still see us referencing the same form items using different types/variables containing the same hardcoded string values. either eliminate one of them or extract tranche name into an enum or union type and use that. I prefer just extracting anyways as these names are used in many places.

File: src/consts.ts [hunk 7]
```diff
@@ -42,90 +43,126 @@ export const dealsRoute: Route = {
...
 };
 
-export const defaultTranches = [oneTrancheStructure, twoTrancheStructure, threeTrancheStructure];
+export const defaultTranches: DefaultTranche[] = [
+	oneTrancheStructure,
+	twoTrancheStructure,
+	threeTrancheStructure,
+];
 
 export const trancheNames = ["super senior", "senior", "mezzanine", "junior"];
```
Comment (RuvendeGroote) [inline src/consts.ts]: replace these values now that we have the enum

File: src/types/tranche.types.ts [hunk 8]
```diff
@@ -7,3 +7,9 @@ export interface Tranche {
 	percentageOfPrincipal: Fraction;
 	percentageOfInterest: Fraction;
 }
+
+export enum TrancheName {
```
Comment (RuvendeGroote) [inline src/types/tranche.types.ts:11]: this enum can be used everywhere we talk about tranches

**Review by RuvendeGroote — CHANGES_REQUESTED** [review]

**Review by RuvendeGroote — APPROVED** [review]

**Pull request merged**